backend/app/controllers/spree/admin/customer_returns_controller.rb

```diff
@@ -31,15 +31,15 @@ def build_resource
       end
 
       def find_resource
-        Spree::CustomerReturn.accessible_by(current_ability, :show).find(params[:id])
+        current_store.customer_returns.accessible_by(current_ability, :show).find(params[:id])
       end
 
       def collection
         parent # trigger loading the order
-        @collection ||= Spree::ReturnItem.
+        @collection ||= current_store.customer_returns.where(id: Spree::ReturnItem.
                         accessible_by(current_ability).
                         where(inventory_unit_id: @order.inventory_units.pluck(:id)).
-                        map(&:customer_return).uniq.compact
+                        map(&:customer_return).uniq.compact.map(&:id))
```

Comment:
Doesn't it return relation of customer returns?
```
current_store.customer_returns.where(id: Spree::ReturnItem.
                        accessible_by(current_ability).
                        where(inventory_unit_id: @order.inventory_units.pluck(:id))
```
If so, we could do something like
```suggestion
                        distinct.ids.compact
```
Compact could probably also be changed to AR query.